integrations/amazon_sagemaker/tests/test_sagemaker.py

```diff
@@ -1,243 +1,175 @@
 import os
-from unittest.mock import Mock
+from unittest.mock import Mock, patch
 
 import pytest
+from botocore.exceptions import BotoCoreError
+from haystack.utils.auth import EnvVarSecret
 from haystack_integrations.components.generators.amazon_sagemaker import SagemakerGenerator
 from haystack_integrations.components.generators.amazon_sagemaker.errors import AWSConfigurationError
 
+mocked_dict = {
+    "type": "haystack_integrations.components.generators.amazon_sagemaker.sagemaker.SagemakerGenerator",
+    "init_parameters": {
+        "model": "model",
+        "aws_access_key_id": {"type": "env_var", "env_vars": ["AWS_ACCESS_KEY_ID"], "strict": False},
+        "aws_secret_access_key": {"type": "env_var", "env_vars": ["AWS_SECRET_ACCESS_KEY"], "strict": False},
+        "aws_session_token": {"type": "env_var", "env_vars": ["AWS_SESSION_TOKEN"], "strict": False},
+        "aws_region_name": {"type": "env_var", "env_vars": ["AWS_DEFAULT_REGION"], "strict": False},
+        "aws_profile_name": {"type": "env_var", "env_vars": ["AWS_PROFILE"], "strict": False},
+        "aws_custom_attributes": {"accept_eula": True},
+        "generation_kwargs": {"max_new_tokens": 10},
+    },
+}
+
+
+@pytest.mark.unit
+@pytest.mark.usefixtures("set_env_variables", "mock_boto3_session")
```

Comment:
I see no benefits in using `usefixtures` in this case. If we wrap all the tests that need them in a class and decorate the class I'd be ok with that.
Otherwise just them as parameters.

Reply:
for some weird reason, having them as parameters fails - some config issue, I suspect, I will look into it